sds/src/secondary_validation/brazilian_cnpj_checksum.rs

```diff
@@ -0,0 +1,84 @@
+use crate::secondary_validation::Validator;
+
+pub struct BrazilianCnpjChecksum;
+
+const BRAZILIAN_CNPJ_LENGTH: usize = 18;
+// XX.XXX.XXX/YYYY-ZZ
+const BRAZILIAN_CNPJ_SEPARATOR_INDICES: &[usize] = &[2, 6, 10, 15];
+
+impl Validator for BrazilianCnpjChecksum {
+    // https://pt.wikipedia.org/wiki/Cadastro_Nacional_da_Pessoa_Jur%C3%ADdica
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        // Check if the length of the ID is correct
+        if regex_match.len() != BRAZILIAN_CNPJ_LENGTH {
+            return false;
+        }
+
+        // Collect digits from match
```

Comment:
You can possible avoid to collect the numbers by computing the checksum on the fly. [Here is an example](https://github.com/DataDog/dd-sensitive-data-scanner/blob/9c72d7031f9172c2aa84a1c6a2b0702e53a06880/sds/src/secondary_validation/nhs_check_digit.rs#L14-L30) of checksums where the scanner does not collect the numbers.